src/module.c

```diff
@@ -10729,14 +10746,47 @@ void moduleCallCommandFilters(client *c) {
         if ((f->flags & REDISMODULE_CMDFILTER_NOSELF) && f->module->in_call) continue;
 
         /* Call filter */
-        f->callback(&filter);
+        if (f->pre_callback != NULL) {
+            f->pre_callback(&filter);
+        }
     }
 
     c->argv = filter.argv;
     c->argv_len = filter.argv_len;
     c->argc = filter.argc;
 }
 
+void moduleCallCommandPostFilters(client *c, long long dirty, int is_failed) {
+    if (listLength(moduleCommandFilters) == 0) return;
+
+    listIter li;
+    listNode *ln;
+    listRewindTail(moduleCommandFilters,&li);
+
+    RedisModuleCommandFilterCtx ctx = {
+        .argv = c->argv,
+        .argv_len = c->argv_len,
+        .argc = c->argc,
+        .c = c,
+        .is_dirty = dirty > 0 ? 1 : 0,
+        .is_succeeded = is_failed > 0 ? 0 : 1
```

Comment:
AFAICT these are always positive. so this condition looks strange and might be misleading.
if we want to make them boolean, we can do `dirty ? 1 : 0`, or `!!dirty`